src/sas/sascalc/dataloader/readers/IgorReader.py

```diff
@@ -39,223 +38,157 @@ class Reader:
     def read(self, filename=None):
         """ Read file """
         if not os.path.isfile(filename):
-            raise ValueError, \
-            "Specified file %s is not a regular file" % filename
-        
-        # Read file
-        f = open(filename, 'r')
-        buf = f.read()
+            raise ValueError("Specified file %s is not a regular "
+                             "file" % filename)
         
-        # Instantiate data object
         output = Data2D()
+
         output.filename = os.path.basename(filename)
         detector = Detector()
-        if len(output.detector) > 0:
-            print str(output.detector[0])
+        if len(output.detector):
+            print(str(output.detector[0]))

```

Comment:
Generally shouldn't have print statements in production code.  Does this show up in the sasview console?  Or only on the command line?  

For sanity purposes, include a searchable string in the output so that we can find the print statements when we want to remove them.  E.g.,`print("igor detector 0: %s"%str(output.detector[0]))`.

Another ticket: make non-gui tests run on python 3.x.